wrapanapi/systems/rhevm.py

```diff
@@ -451,10 +453,12 @@ def add_disk(self, storage_domain=None, size=None, interface='VIRTIO', format=No
         """
         disk_attachments_service = self.api.disk_attachments_service()
         disk_attach = types.DiskAttachment(
-            disk=types.Disk(format=types.DiskFormat(format),
+            disk=types.Disk(name=str(name),
+                            format=types.DiskFormat(format.lower()),
                             provisioned_size=size,
-                            storage_domains=[types.StorageDomain(name=storage_domain)]),
-            interface=getattr(types.DiskInterface, interface),
+                            storage_domains=[types.StorageDomain(name=storage_domain)],
+                            sparse=bool(sparse)),
```

Comment:
it should already be bool

Reply:
it should, but there's no type validation otherwise. I can remove though and just pass through what we get.

Reply:
I like that type validation. It will throw error right there, before performing operations with incorrect values.